File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxi=0
        i,j=0,1
        profit=0
        # while j>=0:
        #     profit=max(profit,0)
        #     profit-=
        for p in range(1,len(prices)):
            j=p
            if prices[j]<prices[i]:
                i=p
            elif prices[j]>prices[i]:
                maxi=max(maxi,prices[j]-prices[i])








        # while i<j:
        #     if prices[j]<prices[i]:
        #         j-=1
        #         continue
        #     elif profit<(prices[j]-prices[i]):
        #         profit=prices[j]-prices[i]
        #         maxi=max(maxi,profit)
        #     i+=1
        #     j-=1
        return maxi

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = Integer.MIN_VALUE, min_buy = Integer.MAX_VALUE;
        
        for (int i = 0; i < prices.length; i++){
            if (prices[i] < min_buy) min_buy = prices[i];
            maxProfit = Math.max(maxProfit, prices[i] - min_buy);
        } 
        return maxProfit;
    }
}